packages/plugin-ext/src/plugin/notebook/notebooks.ts

```diff
@@ -0,0 +1,284 @@
+import { NotebookEditorExtImpl } from './notebook-editor';
+
+export class NotebooksExtImpl implements NotebooksExt {
+
+    private readonly notebookStatusBarItemProviders = new Map<number, theia.NotebookCellStatusBarItemProvider>();
+    private readonly commandsConverter: CommandsConverter;
+
+    private readonly DidChangeActiveNotebookEditorEmitter = new Emitter<theia.NotebookEditor | undefined>();
+    readonly onDidChangeActiveNotebookEditor = this.DidChangeActiveNotebookEditorEmitter.event;
+
+    private DidOpenNotebookDocumentEmitter = new Emitter<theia.NotebookDocument>();
+    onDidOpenNotebookDocument: Event<theia.NotebookDocument> = this.DidOpenNotebookDocumentEmitter.event;
+    private DidCloseNotebookDocumentEmitter = new Emitter<theia.NotebookDocument>();
+    onDidCloseNotebookDocument: Event<theia.NotebookDocument> = this.DidCloseNotebookDocumentEmitter.event;
+
+    private DidChangeVisibleNotebookEditorsEmitter = new Emitter<theia.NotebookEditor[]>();
+    onDidChangeVisibleNotebookEditors = this.DidChangeVisibleNotebookEditorsEmitter.event;
+
+    private readonly documents = new Map<URI, NotebookDocument>();
+    private readonly editors = new Map<string, NotebookEditorExtImpl>();
+    private statusBarCache = new Cache<Disposable>('NotebookCellStatusBarCache');
+
+    private notebookProxy: NotebooksMain;
+
+    constructor(
+        rpc: RPCProtocol,
+        // private editorsAndDocuments: EditorsAndDocumentsExtImpl
+    ) {
+        this.notebookProxy = rpc.getProxy(PLUGIN_RPC_CONTEXT.NOTEBOOKS_MAIN);
+    }
+
+    async $provideNotebookCellStatusBarItems(handle: number, uri: UriComponents, index: number, token: CancellationToken): Promise<NotebookCellStatusBarListDto | undefined> {
+        const provider = this.notebookStatusBarItemProviders.get(handle);
+        const revivedUri = URI.fromComponents(uri);
+        const document = this.documents.get(revivedUri);
+        if (!document || !provider) {
+            return;
+        }
+
+        const cell = document.getCellFromIndex(index);
+        if (!cell) {
+            return;
+        }
+
+        const result = await provider.provideCellStatusBarItems(cell.apiCell, token);
+        if (!result) {
+            return undefined;
+        }
+
+        const disposables = new DisposableCollection();
+        const cacheId = this.statusBarCache.add([disposables]);
+        const resultArr = Array.isArray(result) ? result : [result];
+        const items = resultArr.map(item => typeConverters.NotebookStatusBarItem.from(item, this.commandsConverter, disposables));
+        return {
+            cacheId,
+            items
+        };
+    }
+
+    $releaseNotebookCellStatusBarItems(cacheId: number): void {
+        this.statusBarCache.delete(cacheId);
+    }
+
+    // --- serialize/deserialize
+
+    private _handlePool = 0;
+    private readonly _notebookSerializer = new Map<number, theia.NotebookSerializer>();
```

Comment:
The only time we use the `_variableName` pattern in Theia is when we have a property function.